crowd.py

```diff
@@ -628,7 +628,7 @@ def get_memberships(self):
             memberships[group] = {u'users': users, u'groups': groups}
         return memberships
 
-    def search(self, entity_type, property_name, search_string):
+    def search(self, entity_type, property_name, search_string, start_index=0, max_results=99999):
```

Comment:
@hdost ,
Thank you for the feedback. 1000 was my original plan and I'm happy to change from 99999 to 1000. However, please consider the following:

1. Atlassian indicated default results set size maximums may change without notice. If Atlassian increases default size, python-crowd would be masking the change with a lower default value. If a tool using python-crowd is not supplying a max_results size while assuming they will receive the default max_results set by the server, they would need to alter their code to declare a specific max_results value. Since python-crowd has not previously supported max-results, existing usage may be relying upon the server-side setting to control this behavior.

2. If a server is configured with the Atlassian default of 1000 results, the max_results query for 99999 will be ignored by the server and only the 1000 results will be returned. 99999 seems high enough to allow server enforcement any practical max_results settings.

3. Less important IMO, but convention within python-crowd method get_nested_group_users defaults max-results to 99999. May not be worth following this convention, but mentioning for completeness.

If recommended change to 1000 stands, I'm happy to change the default value.

Reply:
@potus98 That sounds reasonable.